src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -510,26 +586,32 @@ public async Task ExecuteClusterTransaction(TransactionOperationContext context)
             catch (Exception e)
             {
                 if (_logger.IsInfoEnabled)
-                {
                     _logger.Info($"Failed to execute cluster transaction batch (count: {batch.Count}), will retry them one-by-one.", e);
-                }
-                await ExecuteClusterTransactionOneByOne(batch);
+
+                if (await ExecuteClusterTransactionOneByOne(batch.GetData()))
+                    NotifyIndexIfNeeded();
                 return;
             }
-            foreach (var command in batch)
+            foreach (var command in batch.GetData())
             {
                 OnClusterTransactionCompletion(command, mergedCommands);
             }
+            
+            NotifyIndexIfNeeded();
+
+            void NotifyIndexIfNeeded()
```

Comment:
let's pass the raftIndex here and move this method outside

Reply:
done